arches_lingo/src/arches_lingo/components/scheme/report/SchemeNamespace.vue

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+const OPEN_EDITOR = "openEditor";
+</script>
+<script setup lang="ts">
+import { onMounted, ref } from "vue";
+import { useRoute } from "vue-router";
+import { useGettext } from "vue3-gettext";
+
+import SchemeReportSection from "@/arches_lingo/components/scheme/report/SchemeSection.vue";
+import NonLocalizedString from "@/arches_lingo/components/generic/NonLocalizedString.vue";
+import {
+    fetchSchemeNamespace,
+    updateSchemeNamespace,
+} from "@/arches_lingo/api.ts";
+import type {
+    DataComponentMode,
+    SchemeNamespaceUpdate,
+    SchemeNamespace,
+} from "@/arches_lingo/types";
+
+const { $gettext } = useGettext();
+
+const schemeNamespace = ref<SchemeNamespace>();
+const route = useRoute();
+defineProps<{
+    mode?: DataComponentMode;
+}>();
+defineEmits([OPEN_EDITOR]);
+
+const save = async () => {
+    await updateSchemeNamespace(
+        route.params.id as string,
+        schemeNamespace.value as SchemeNamespace,
+    );
+};
+const getSectionValue = async () => {
+    const response = await fetchSchemeNamespace(route.params.id as string);
+    schemeNamespace.value = response;
+};
+
+defineExpose({ save, getSectionValue });
+
+onMounted(async () => {
+    getSectionValue();
+});
+
+const onNamespaceNameUpdate = (val: string) => {
```

Comment:
I'd say declare all functions with `function foo()` declarations, and put them on the bottom of the script tag so they're not mixed in with `onMounted` and `defineExpose`. There's an ordering of stuff in the script tag that I haven't promulgated yet, but for the time being IMO we should still be writing for readability.

Reply:
fair enough